File: 1395. Count Number of Teams/dp.js
/**
 * @param {number[]} rating
 * @return {number}
 */
 var numTeams = function(rating) {
        
    const n = rating.length;
    const smaller = [...Array(n)].fill(0), greater = [...Array(n)].fill(0);
    let res = 0;
    // Using i as last integer in the subset
    for(let i = 0; i < n; i++){
        
        // Check all element on its left
        for(let j = 0; j < i; j++){
            // For element smaller than i, j, a new subset can be form with any number smaller than j
            if(rating[j] < rating[i]){
                smaller[i]++;
                res += smaller[j];
                
            }
            // For element greater than i, j, a new subset can be form with any number greater than j
            if(rating[j] > rating[i]){
                greater[i]++;
                res += greater[j];
            }
        }
    }
    
    return res;
    
};

File: 1395. Count Number of Teams/dp.py
class Solution:
    def numTeams(self, rating: List[int]) -> int:
        res = 0;
        n = len(rating);
        smaller, greater = [0] * n, [0] * n;

        for i in range(0, n):
            for j in range(0, i):
                # Check if j is greater than i
                if rating[j] > rating[i]:
                    greater[i] += 1; # The element j is greater than i
                    res += greater[j];
                
                # Check if j is smaller than i
                if rating[j] < rating[i]:
                    smaller[i] += 1;
                    res += smaller[j];
        
        return res;
        